design/ceph/ceph-update-osds-in-parallel.md

```diff
@@ -0,0 +1,113 @@
+# Updating OSDs in parallel
+**Targeted for v1.6**
+
+## Background
+In clusters with large numbers of OSDs, it can take a very long time to update all of the OSDs. This
+occurs on updates of Rook and Ceph both for major as well as the most minor updates. To better
+support large clusters, Rook should be able to update (and upgrade) multiple OSDs in parallel.
+
+
+## High-level requirements
+
+### Different update strategies
+Users will likely wish to specify different upgrade strategies. For many clusters, updating all OSDs
+on a node may be sufficient. For portable OSDs on PVCs, an OSD may not be associated directly with a
+node.
+
+Users may wish to update all OSDs in a particular failure domain at once, updating all OSDs in a
+domain before moving to the next domain. Some examples of failure domains are:
+ - by OSD (equivalent to the existing strategy)
+ - by node/host
+ - by rack
+ - by data center room
+ - by availability zone
+ - and many others
+
+> **Open question:** Will it be useful for users to try to update OSDs **across** failure domains?
```

Comment:
> Will it be useful for users to try to update OSDs **across** failure domains?

I'd expect to update OSDs within a failure domain. If you had OSDs in two failure domains down at the same time, data would become unavailable.

> update some number of OSDs in each failure domain instead of updating all OSDs within a
> given failure domain?

I can imagine the admin wanting to proceed with some percentage of a failure domain at a time instead of blasting all OSDs in a failure domain at the same time.

Reply:
Makes sense. Removed.